### PR TITLE
Simulated Platform to aid development

### PR DESCRIPTION
**Pull request by bsutton**
I've created a simulated platform that mimics any of the existing platforms (at a very basic level) to allow development on a standard PC platform.

To use the simulator you need two environment variables:
The standard PI4J platform statement MUST point to the simulator:
`PI4J_PLATFORM=Simulated`

A second environment variable that defines the platform that is to be simulated.
`SimulatedPlatform=<Real Platform's Name>`

e.g.
`SimultatedPlatform=RaspberryPi GPIO Provider`

### CODE REVIEW

**Comment by bsutton** [comment]
just wondering if there is any chance of getting this merged?
I'm using it actively in a project using digital output pins and it works great.
I figured if we can get this into the core project the hopefully over time we will get additional contributions expanding the set of features that can be simulated.

**Comment by natdan** [comment]
Of course. I am sorry for being so slow. I'll try to go through the code until weekend unless @savageautomate doesn't do it before me!

File: pi4j-core/src/main/java/com/pi4j/io/gpio/SimulatedGpioProvider.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ * 
+ * @author bsutton
+ *
+ */
+public class SimulatedGpioProvider extends GpioProviderBase implements GpioProvider {
+	
+	// We use the name of the platform that we are simulating.
+    public static  String NAME;
+	
+	public SimulatedGpioProvider()
```
Comment (natdan) [inline pi4j-core/src/main/java/com/pi4j/io/gpio/SimulatedGpioProvider.java]: Could we, please, just use 4 spaces? (are below tabs?)

**Comment by natdan** [comment]
Oh, I didn't know it is that simple! I've put a couple of styling comments (spaces/tabs/alignment) - nothing serious...

**Comment by bsutton** [comment]
Sounds great

**Comment by bsutton** [comment]
as requested, tabs replaced with spaces.

**Comment by marklister** [comment]
How does one get around `SEVERE: Unable to load [libpi4j.so] using path: [/lib/raspberrypi/dynamic/libpi4j.so]` on the dev platform?

**Comment by savageautomate** [comment]
Which version of Pi4J are you running.  That SO library file should be embedded in the `pi4j-core.jar` and extracted at runtime.

**Comment by marklister** [comment]
I'm using 1.3-SNAPSHOT.  
The error I get is:
`SEVERE: Unable to load [libpi4j.so] using path: [/lib/simulated/static/libpi4j.so]`

In the jar I can see a `lib/raspberrypi/dynamic/libpi4j.so`

**Comment by marklister** [comment]
I set the enviroment variables manually and ran sbt from command line to get it to look for simulated platform

`declare -x PI4J_PLATFORM="Simulated"`
`declare -x SimultatedPlatform="BananaPi GPIO Provider"`

Using the systemInfo util:
```server ----------------------------------------------------
server PLATFORM INFO
server ----------------------------------------------------
server Platform Name     :  Simulated
server Platform ID       :  simulated
```

I've tried it as a RaspberryPi GPIO Provider too.

**Pull request merged**